File: dotnet/aws/Mcma.Aws.DynamoDb/DynamoDbTableOptions.cs
using System.Collections.Generic;

namespace Mcma.Aws.DynamoDb
{
    public class DynamoDbTableOptions
    {
        public bool? ConsistentGet { get; set; }
        
        public bool? ConsistentQuery { get; set; }
        
        public Dictionary<string, string> TopLevelAttributeMappings { get; set; } = new Dictionary<string, string>();
        
        public Dictionary<string, string> IndexNameMappings { get; set; } = new Dictionary<string, string>();
    }
}

File: aws/dynamodb/lib/dynamo-db-table-options.ts
import { CustomQuery } from "@mcma/data";
import { QueryInput } from "@aws-sdk/client-dynamodb";

type AttributeValueRetriever = (partitionKey: any, sortKey: any, resource: any) => any;

export type CustomQueryFactory = (customQuery: CustomQuery) => QueryInput;
export type CustomQueryRegistry = { [key: string]: CustomQueryFactory };

export interface DynamoDbTableOptions {
    consistentGet?: boolean;
    consistentQuery?: boolean;
    topLevelAttributeMappings?: { [key: string]: AttributeValueRetriever };
    customQueryRegistry?: CustomQueryRegistry;
}
